File: java/socketd/src/main/java/org/noear/socketd/transport/client/ClientConnectHandler.java
package org.noear.socketd.transport.client;

import org.noear.socketd.transport.core.ChannelInternal;

import java.io.IOException;

/**
 * 客户端连接处理器
 *
 * @author noear
 * @since 2.4
 */
@FunctionalInterface
public interface ClientConnectHandler {
    /**
     * 连接
     *
     * @param connector 连接器
     */
    ChannelInternal clientConnect(ClientConnector connector) throws IOException;
}

File: python/socketd/socketd/transport/client/ClientConnectHandler.py
from typing import Callable

from socketd.transport.client.ClientConnector import ClientConnector
from socketd.transport.core.ChannelInternal import ChannelInternal

ClientConnectHandler = Callable[[ClientConnector], ChannelInternal]

def ClientConnectHandlerDefault(connector:ClientConnector):
    return connector.connect()
